### ISSUE TITLE
[BUG] iPhone 14 family not recognized

### ISSUE THREAD

**Issue by gabriel-pq** [microsoft/playwright#26772]
### System info
- Playwright Version: [v1.37.1]
- Operating System: [macOS 13.5.1]
- Browser: N/A
- Other info: The scenario is valid and similar for each device from iPhone 14 family present here: https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/server/deviceDescriptorsSource.json

**Config file**

```js
// playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  projects: [
    {
    name: 'iPhone 14 - iOS 16 - Safari',
    use: {
      ...devices['iPhone 14'],
    },
  },
});
```

**Steps**

The scenario is super simple!

1. Add any iPhone 14 in the config file
2. Spin a test that visits a page (e.g. https://playwright.dev/) with the PLAYWRIGHT_HEADLESS=0

**Expected**

I would expect to see a mobile view of the iPhone 14.

**Actual**

![image](https://github.com/microsoft/playwright/assets/85311517/0760ab33-f580-45a9-9875-a440b1532533)

The behavior is similar to when I put a random config:
![image](https://github.com/microsoft/playwright/assets/85311517/d292baa6-db6f-4441-91c9-5ef4113dd77e)

I've seen this feature [[Feature] Add iPhone 14 family](https://github.com/microsoft/playwright/issues/24214)  being closed and that's why I opened this bug!

**Comment by mxschmitt** [comment]
iPhone 14 Family is not released yet, you can either wait for 1.38 or use [Canary](https://playwright.dev/docs/canary-releases).

**Comment by gabriel-pq** [comment]
Thanks!
